diff -urN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2012-05-24 08:16:45.000000000 -0400
+++ b/CMakeLists.txt	2012-05-24 08:16:14.000000000 -0400
@@ -238,3 +238,11 @@
   cxx_executable(gtest_xml_output_unittest_ test gtest)
   py_test(gtest_xml_output_unittest)
 endif()
+
+install(TARGETS gtest
+	ARCHIVE DESTINATION ${VGLIB_BINARY_DIR}/lib
+)
+install(DIRECTORY include/
+	DESTINATION ${VGLIB_BINARY_DIR}/include
+)
+
